posts/inside-rust/2021-03-05-async-vision-doc.md

```diff
@@ -0,0 +1,98 @@
+[dt]: https://rust-lang.github.io/wg-async-foundations/vision/tenets.html
+[cc]: https://rust-lang.github.io/wg-async-foundations/vision/characters.html
+[dd]: https://rust-lang.github.io/wg-async-foundations/design_docs.html
+[gf]: https://forms.gle/YKNniGhaNXBhmjXNA
+[gr]: https://is.gd/T6TadC
+[z]: https://rust-lang.zulipchat.com/
+[wgz]: https://rust-lang.zulipchat.com/#streams/187312/wg-async-foundations
+[amb]: https://rust-lang.github.io/wg-async-foundations/vision/shiny_future.html#think-big----too-big-if-you-have-to
+[cok]: https://en.wikipedia.org/wiki/Curse_of_knowledge
+[workarounds]: https://rust-lang.github.io/async-book/07_workarounds/01_chapter.html
+[stabilized]: https://blog.rust-lang.org/2019/11/07/Async-await-stable.html
+
+Async-await first [stabilized] in November 2019, and today people are building all kinds of applications using it. The [Async Foundations Working Group][wg] believes Rust can become one of the most popular choices for building distributed systems, ranging from embedded devices to foundational cloud services.
+
+Regardless of their application, we want all developers to love using Async Rust. For that to happen, we need to move Async Rust beyond the "MVP" state it's in today and make it accessible to everyone.
+
+We are launching a collaborative effort to build a shared [vision document][vd] for Async Rust. **Our goal is to engage the entire community in a collective act of the imagination:** how can we make the end-to-end experience of using Async I/O not only a pragmatic choice, but a _joyful_ one?
+
+If you'd like to be a part of that, good news: We need your help! As a first step, we want to encourage people to write about their experiences using Async Rust. Read on for more.
+
+### Involving the whole community
+
+Building a truly great experience for Async Rust takes more than just `async fn`. We need to think about the end-to-end experience, from first learning Async Rust, to building an application, to deployment and maintenance.
+
+The challenge for us is to drive that holistic product experience while retaining the loosely coupled, exploration-oriented ecosystem we have built. Async Rust was intentionally designed not to have a "one size fits all" mindset, and we don't want to change that.
+
+However, we _do_ need a forum where we can coordinate on the overall experience we are shooting for. This is where the [vision doc][vd] comes in.
+
+### To know where you're going, you have to know where you are
+
+Our first goal for the [vision document][vd] is to describe what it's like to use Async Rust today. We've created a [cast of characters][cc] that represent the different backgrounds and experiences that people bring to bear when learning and using Async Rust:
+
+* **Alan**, the full-stack developer working across many companies,
+* **Grace**, the principal engineer hacking on a data storage service,
+* **Niklaus**, the developer building generic Rust libraries and frameworks;
```

Comment:
nit: The first two items use commas at the end, but the last one uses a semicolon.

Reply:
That was my copy edit; the semicolon is because it's the last item in the list (takes the place of ", and").

I don't actually care that much about this, though :)

Reply:
> That was my copy edit; the semicolon is because it's before the last item in the list (takes the place of ", and").

Hmm, I've never seen that style before! I suspect most people will assume it's a typo, but it's not a big deal. If you do decide to change it, I would suggest

- using all commas,
- replacing the semicolon with ", and", or
- removing the punctuation altogether